### PR TITLE
Add tests for UnmanagedMemoryAccessor struct methods

### PR DESCRIPTION
**Pull request by ianhays**
Progress towards https://github.com/dotnet/corefx/issues/9465

exposure of the methods was completed by @alexperovich when he moved UnmanagedMemoryStream/Accessor to corelib in https://github.com/dotnet/corefx/pull/12391 & https://github.com/dotnet/coreclr/pull/7418

@stephentoub @joperezr @alexperovich

### CODE REVIEW

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Uma_ReadWriteStruct
+    {
+        private const int UmaTestStruct_DisalignedSize = 2 * sizeof(int) + 2 * sizeof(bool) + sizeof(char);
```
Comment (stephentoub) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs]: Nit: disaligned => unaligned

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            public bool bool1;
+            public int int2;
+            public char char1;
+            public bool bool2;
+        }
+
+        private struct UmaTestStruct_ContainsReferenceType
+        {
+            public object referenceType;
+        }
```
Comment (stephentoub) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs]: Consider also having a generic struct

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStructArray.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            public bool bool1;
+            public int int2;
+            public char char1;
+            public bool bool2;
+        }
+
+        private struct UmaTestStruct_ContainsReferenceType
+        {
+            public object referenceType;
+        }
```
Comment (stephentoub) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStructArray.cs]: Could we put these helper types in a shared file or a base class or somewhere so we don't have to redefine them in multiple places?

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs (hunk 1 above)
Comment (ianhays) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs]: thanks.

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs (hunk 2 above)
Comment (ianhays) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStruct.cs]: Good idea.

File: src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStructArray.cs (hunk 3 above)
Comment (ianhays) [inline src/System.IO.UnmanagedMemoryStream/tests/Uma.ReadWriteStructArray.cs]: Yup!

File: src/System.IO.UnmanagedMemoryStream/tests/System.IO.UnmanagedMemoryStream.Tests.csproj [hunk 4]
```diff
@@ -15,13 +15,11 @@
...
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-  </PropertyGroup>
-  <PropertyGroup>
-  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' " />
   <ItemGroup>
+    <Compile Include="Uma.ReadWriteStruct.cs" Condition="'$(TargetGroup)' == ''" />
```
Comment (ianhays) [inline src/System.IO.UnmanagedMemoryStream/tests/System.IO.UnmanagedMemoryStream.Tests.csproj]: Done.

**Comment by ianhays** [comment]
Thanks for the feedback folks! I pushed an update responding to it all and will merge when CI is green.

**Pull request merged**